rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystem.java

```diff
@@ -85,7 +86,6 @@ private static void initialize(Configuration conf, ClusterType clusterType)
       if (initialized.get()) {
         return;
       }
-
       initializeStats(conf);
```

Comment:
I don't think stats should be initialized on caching FS initialization. In Presto we create more than one `CachingFileSystem` (with different configs: one with cache enabled, one with cache disabled, one for bookeeper/filedownloader). Since stats are dependent on configuration, we are not really sure which config would be used for metrics.

It seems that `setLocalBookKeeper` should be the method initializing/reinitializing stats.

Reply:
We setLocalBookKeeper version will always create the stats first in embedded mode as we do not open up presto for traffic before BookKeeper is initialized. For safety, we can remove the check later in setLocalBookKeeper and ensure we always override stats through it.

The part about initializing stats via FS need to stay because in standalone mode that is the only first point where `Configuration` object is accessible.

Reply:
> The part about initializing stats via FS need to stay because in standalone mode that is the only first point where Configuration object is accessible.

Maybe stats could be initialized in CachingFS only in non-embedded mode? This could be checked from passed `Configuration`